### PR TITLE
experimentalEnum config

### PR DESCRIPTION
**Pull request by nedredmond**
## Summary:
Adds the ability to replace the default string literal union types with flow enums, which are exported for use within the application.

Issue: LP-11757

## Test plan:
Thoroughly unit tested. Add `graphql-flow.config.js` with `experimentalEnums` enabled to test.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -141,3 +146,14 @@ export const documentToFlowTypes = (
     }
     return result;
 };
+
+function processExtraTypes(types: {[key: string]: BabelNode}): {
```
Comment (jaredly) [inline src/index.js:150]: maybe `codegenExtraTypes`? to be a little more descriptive

**Review by jaredly — APPROVED** [review]
please update the readme for this one as well!

File: src/generateTypeFiles.js [hunk 2]
```diff
@@ -62,40 +63,56 @@ export const generateTypeFileContents = (
...
+                    (fileContents += `\n\nexport type ${name} = ${extraTypes[name]};`),
+            );
+            const enumNames = Object.keys(experimentalEnums);
+            if (options.experimentalEnums && enumNames.length) {
+                // TODO(somewhatabstract, FEI-4172): Update to fixed eslint-plugin-flowtype
+                // and remove this disable.
+                fileContents += `\n\n/* eslint-disable no-undef */`;
+                enumNames.forEach((name) =>
+                    (fileContents += `\nexport ${experimentalEnums[name]};\n`).replace(
+                        / +/,
```
[the remark this answers is not in the document]
Comment (nedredmond) [reply src/generateTypeFiles.js:101]: Actually, this is leftover from my multiline string literal, where MANY left indents were added. I believe I can remove this. Thanks!

**Comment by nedredmond** [comment]
The parent pull-request ([#40](https://github.com/Khan/graphql-flow/pull/40)) has been merged into `main`, but this branch (`LP-11757-enums`) now has conflicts with the new base branch. These conflicts must be resolved before checks can complete on this pull-request.

**Pull request merged**